src/components/Interactive.d.ts

```diff
@@ -0,0 +1,12 @@
+import { CSSProperties, PropsWithChildren } from 'react'
+
+
+export type InteractiveProps<Tag extends keyof JSX.IntrinsicElements> = PropsWithChildren<{
```

Comment:
`@types/react` defines this. `T extends keyof JSX.IntrinsicElements` means that this type parameter must be an HTML tag name.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9a9b23e4da5d9e8d7402047512e7985df4b50f62/types/react/index.d.ts#L3140-L3320